interfaces/repo.go

```diff
@@ -691,42 +691,42 @@ func (r *Repository) connected(snapName, plugOrSlotName string) ([]ConnRef, erro
 
 	if plug, ok := r.plugs[snapName][plugOrSlotName]; ok {
 		for slotInfo := range r.plugSlots[plug] {
-			connRef := *NewConnRef(plug, slotInfo)
+			connRef := NewConnRef(plug, slotInfo)
 			conns = append(conns, connRef)
 		}
 	}
 
 	if slot, ok := r.slots[snapName][plugOrSlotName]; ok {
 		for plugInfo := range r.slotPlugs[slot] {
-			connRef := *NewConnRef(plugInfo, slot)
+			connRef := NewConnRef(plugInfo, slot)
 			conns = append(conns, connRef)
 		}
 	}
 
 	return conns, nil
 }
 
-func (r *Repository) Connections(snapName string) ([]ConnRef, error) {
+func (r *Repository) Connections(snapName string) ([]*ConnRef, error) {
 	r.m.Lock()
 	defer r.m.Unlock()
 
 	if snapName == "" {
 		snapName, _ = r.guessCoreSnapName()
 		if snapName == "" {
-			return nil, fmt.Errorf("internal error: cannot obtain core snap name while computing connections")
+			return nil, fmt.Errorf("snap name is empty")
```

Comment:
Is this change warranted?

Reply:
Nope, it's my error in merging, thanks for spotting!